xarray/core/resample.py

```diff
@@ -0,0 +1,344 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from scipy.interpolate import interp1d
+
+from . import ops
+from .combine import merge
+from .groupby import DataArrayGroupBy, DatasetGroupBy, GroupBy, _dummy_copy
+from .utils import maybe_wrap_array
+
+RESAMPLE_DIM = '__resample_dim__'
+
+
+class Resample(GroupBy):
```

Comment:
Let's not inherit from `GroupBy` here. We already inherit from `DataArrayGroupBy` and `DatasetGroupBy` in the resample subclasses, so this just makes the subclass hierarchy more complicated.